apps/system/js/wrapper_factory.js

```diff
@@ -96,37 +96,48 @@
       // Use fake origin for named windows in order to be able to reuse them,
       // otherwise always open a new window for '_blank'.
       var origin = null;
-      if (name == '_blank') {
-
-        // If we already have a browser and we receive an open request,
-        // display it in the current browser frame.
-        var activeApp = Service.query('AppWindowManager.getActiveWindow');
-        if (activeApp && (activeApp.isBrowser() || activeApp.isSearch())) {
-          activeApp.navigate(url);
-          return;
-        }
+      switch (name) {
+        case 'pinned':
```

Comment:
I think this string should be prefixed by a "_" as it's a special case like "_blank". I wonder whether "_samescope" might be a better name as that's really what it does? It will open in any window with the same scope. "pinned" is a bit specific to the UI of Firefox OS.